frames/src/main/java/com/checkout/frames/view/InternalButton.kt

```diff
@@ -0,0 +1,149 @@
+)
+
+@SuppressWarnings("MagicNumber")
+@Preview(showBackground = true, name = "OutlineButtonPreview")
+@Composable
+private fun OutlineButtonPreview() {
+    val containerMapper = ContainerStyleToModifierMapper()
+    val styleMapper = ButtonStyleToInternalViewStyleMapper(
+        containerMapper,
+        TextLabelStyleToViewStyleMapper(containerMapper)
+    )
+    val stateMapper = ButtonStyleToInternalStateMapper(
+        TextLabelStyleToStateMapper(
+            imageMapper = ImageStyleToComposableImageMapper()
+        )
+    )
+
+    val style = ButtonStyle(
+        containerColor = 0x00000000,
+        disabledContainerColor = 0xFFCCCCCC,
+        contentColor = 0xFF0B5FF0,
+        disabledContentColor = 0xFFFFFF00,
+        shape = Shape.RoundCorner,
+        cornerRadius = CornerRadius(8),
+        borderStroke = BorderStroke(1, 0xFF8099FD),
+        contentPadding = Padding(start = 16, end = 16, top = 8, bottom = 8),
+        textStyle = DefaultTextLabelStyle.title().copy(
+            textStyle = TextStyle(size = 15, font = Font.SansSerif),
+            text = "Add billing address",
+            trailingIconStyle = ImageStyle(
+                image = R.drawable.cko_ic_caret_right,
+                tinColor = 0xFF0B5FF0,
+                height = 12,
+                padding = Padding(start = 12)
+            )
+        ),
+        containerStyle = ContainerStyle(margin = Margin(16))
+    )
+
+    val state = stateMapper.map(style)
+
+    InternalButton(
+        styleMapper.map(style).apply { textStyle.textMaxWidth = true },
+        state
+    ) { state.isEnabled.value = !state.isEnabled.value }
+}
+
+@SuppressWarnings("MagicNumber")
+@Preview(showBackground = true, name = "OutlineButtonPreview")
+@Composable
+private fun ButtonPreview() {
+    val containerMapper = ContainerStyleToModifierMapper()
+    val styleMapper = ButtonStyleToInternalViewStyleMapper(
+        containerMapper,
+        TextLabelStyleToViewStyleMapper(containerMapper)
+    )
+    val stateMapper = ButtonStyleToInternalStateMapper(
+        TextLabelStyleToStateMapper(
+            imageMapper = ImageStyleToComposableImageMapper()
+        )
+    )
+
+    val style = ButtonStyle(
+        containerColor = 0xFF0B5FF0,
+        disabledContainerColor = 0xFFCCCCCC,
+        contentColor = 0xFFFFFFFF,
+        disabledContentColor = 0xFF000000,
+        shape = Shape.RoundCorner,
+        cornerRadius = CornerRadius(8),
+        contentPadding = Padding(start = 16, end = 16, top = 8, bottom = 8),
+        textStyle = DefaultTextLabelStyle.title().copy(
+            textStyle = TextStyle(size = 15, font = Font.SansSerif),
+            text = "Save"
+        ),
+        buttonElevation = ButtonElevation(defaultElevation = 8),
+        containerStyle = ContainerStyle(margin = Margin(16))
+    )
+
+    val state = stateMapper.map(style)
+
+    InternalButton(
+        styleMapper.map(style),
+        state
+    ) { state.isEnabled.value = !state.isEnabled.value }
```

Comment:
Not sure for me, how we can show the default enabled colour when the user released the button.